workshops/util.py

```diff
@@ -227,6 +235,13 @@ def create_uploaded_persons_tasks(data):
             except ObjectDoesNotExist as e:
                 raise ObjectDoesNotExist('{0} (for {1})'.format(str(e), row))
 
+    for event in events:
+        # count attendees
+        learners = event.task_set.filter(role__name='learner').count()
+        # update attendees count if number of learner tasks > event.attendance
+        Event.objects.filter(pk=event.pk, attendance__lt=learners) \
+                     .update(attendance=learners)
```

Comment:
No, it doesn't - `attendance` needs to be numeric (ie. non-null). You think it should work for `attendance_isnull=True`?